### DISCUSSION TITLE
Seeking a React App Code Example for a Viewer Layout Similar to RCSB's 3D View

### DISCUSSION THREAD

**Discussion by idanb-droid** [molstar/molstar#761] [Q&A] [answered]
Hello everyone,

I'm currently working on a React app using JavaScript (not TypeScript) and am seeking a code example to implement a viewer layout similar to the RCSB 3D View found here: [https://www.rcsb.org/3d-view/7VYJ](https://www.rcsb.org/3d-view/7VYJ). If anyone has experience or resources they can share, I would greatly appreciate your assistance.

Thank you in advance!

**Comment by dsehnal** [comment, marked as answer]
Their viewer is available here https://github.com/molstar/rcsb-molstar

**Comment by idanb-droid** [comment]
Thank you for your reply dsehnal, 
I'm sorry, I might be missing it, but is there any documentation? or a documentation on how to use it as a component in react?

**Reply by dsehnal** [reply]
http://molstar.org/docs/plugin/#plugincontext-with-built-in-react-ui

If you dig in the code a little you can just instantiate the react ui component directly in your react app too without having to create the dev wrapper with ref.

**Reply by idanb-droid** [reply]
I tried this one before, but I didn't find a way to make the same layout of RCSB with this method.
